content/news/2022-03-20-release-7-highlights.md

```diff
@@ -0,0 +1,96 @@
++++
+title = "Release 7 Highlights"
+date = 2022-03-20T03:43:00Z
+type = "post"
+description = "The flashiest changes of v7: the health-check flag, experimental DAP support, and an overhauled tree-sitter grammar system."
+in_search_index = true
++++
+
+Ranging from small quality-of-life improvements and fixes to large features
+and refactors, Helix 7 brings some exciting changes.
```

Comment:
Helix 7?

Reply:
There's no point in keeping the `0.` (https://0ver.org/). Might make more sense to bump to `1.0` though?

"If both you and someone you don't know use your project seriously, then use a serious version." / "If your software is being used in production, it should probably already be 1.0.0" We'll either number each release similar to browser releases, or use calendar versioning (https://calver.org/).

Reply:
https://0ver.org/about.html#what-to-do

Reply:
`1.0` makes sense to me 👍

having `.0` might come in handy if we ever need to make a patch release

Reply:
I think `1.0` now is still too new for helix given that there are no plugins and stuff yet so users should expect stuff to have breaking changes, so I guess `0.` is good for now.

Reply:
There's always "one more thing". Plugins can easily be 2.0.


> How do I know when to release 1.0.0?
>
> If your software is being used in production, it should probably already be 1.0.0. If you have a stable API on which users have come to depend, you should be 1.0.0. If you’re worrying a lot about backwards compatibility, you should probably already be 1.0.0.

also

> [0ver.org](https://0ver.org/) may list some auspicious names, but it's not a very good look. If you're having trouble picking a version, or are **stuck asymptotically approaching an "actual" release**, do yourself a favor and slap a [CalVer](https://calver.org/) on it.

(emphasis mine)

I'd be ok with a `YY.0M(.MICRO)` which was my aim before release but there were some folks complaining about calver around that time. Semver doesn't really mean much in applications since every new feature can be a "major (breaking) change" and calver works great for quick iterations (rust-analyzer uses it too)

Reply:
I would be ok with CalVer too. Seems like a good choice if we're aiming to do new releases regularly

Reply:
Nit: should we use `YYYY.MM(.MICRO)` instead ? Seeing `v22` at first really left me confused as to how it jumped so far so fast.